config/config.go

```diff
@@ -62,26 +87,173 @@ func TryLoadRootConfig(fromdir string) (cfg hcl.Config, configpath string, found
 		}
 		fromdir = parent
 	}
-	return hcl.Config{}, "", false, nil
+	return nil, "", false, nil
 }
 
-// IsStack returns true if the given directory is a stack, false otherwise.
-func IsStack(rootdir, dir string) (bool, error) {
+// LoadTree loads the whole hierarchical configuration from cfgdir downwards
+// using rootdir as project root.
+func LoadTree(rootdir string, cfgdir string) (*Tree, error) {
+	return loadTree(rootdir, cfgdir, nil)
+}
+
+// RootDir is the configuration rootdir.
+func (tree *Tree) RootDir() string {
+	return tree.rootdir
+}
+
+// IsStack tells if the tree is a stack.
+func (tree *Tree) IsStack() bool {
+	return tree.Root.Stack != nil
+}
+
+// Stacks returns the stack nodes from the tree.
+func (tree *Tree) Stacks() []*Tree {
+	var stacks List
+	if tree.IsStack() {
+		stacks = append(stacks, tree)
+	}
+	for _, children := range tree.Children {
+		stacks = append(stacks, children.Stacks()...)
+	}
+
+	stacks.Sort()
+	return stacks
+}
+
+// Lookup a node from the tree.
```

Comment:
Maybe we should make it explicit on the doc that it will ignore the root node.

Reply:
it doesn't skip the root node, the old code had a misleading comment, I just changed it.
thanks